e2e/dashboards-suite/dashboard-restore.spec.ts

```diff
@@ -0,0 +1,55 @@
+import testDashboard from '../dashboards/TestRestoreDashboard.json';
+import { e2e } from '../utils';
+
+describe('Dashboard restore', () => {
+  beforeEach(() => {
+    e2e.flows.login(Cypress.env('USERNAME'), Cypress.env('PASSWORD'));
+  });
+
+  it('Deleted dashboards appear in Recently Deleted', () => {
+    e2e.flows.importDashboard(testDashboard, 1000, true);
+
+    e2e.pages.Dashboards.visit();
+
+    // Delete dashboard
+    deleteDashboard();
+
+    // Dashboard should appear in Recently Deleted
+    e2e.pages.RecentlyDeleted.visit();
+    e2e.pages.Search.table.row('E2E Test - Restore Dashboard').should('exist');
+
+    // Restore dashboard
+    e2e.pages.Search.table.row('E2E Test - Restore Dashboard').find('[type="checkbox"]').click({ force: true });
+    cy.contains('button', 'Restore').click();
+    cy.contains('p', 'This action will restore 1 dashboard.').should('be.visible');
+    e2e.pages.ConfirmModal.delete().click();
+
+    // Dashboard should appear in Browse
+    e2e.pages.Dashboards.visit();
+    e2e.pages.BrowseDashboards.table.row('E2E Test - Restore Dashboard').should('exist');
+
+    // Delete dashboard
+    deleteDashboard();
+
+    // Permanently delete dashboard
+    e2e.pages.RecentlyDeleted.visit();
+    e2e.pages.Search.table.row('E2E Test - Restore Dashboard').find('[type="checkbox"]').click({ force: true });
+    cy.contains('button', 'Delete permanently').click();
+    cy.contains('p', 'This action will delete 1 dashboard.').should('be.visible');
+    e2e.flows.confirmDelete();
+
+    // Dashboard should not appear in Recently Deleted or Browse
+    e2e.pages.Search.table.row('E2E Test - Restore Dashboard').should('not.exist');
+
+    e2e.pages.Dashboards.visit();
+    e2e.pages.BrowseDashboards.table.row('E2E Test - Restore Dashboard').should('not.exist');
+  });
+});
+
+const deleteDashboard = () => {
+  e2e.pages.BrowseDashboards.table.row('E2E Test - Restore Dashboard').find('[type="checkbox"]').click({ force: true });
+
+  cy.contains('button', 'Delete').click();
+  e2e.flows.confirmDelete();
```

Comment:
Is there an opportunity to check whether the Toast is showing the right text?
<img width="331" alt="Screenshot 2024-09-04 at 11 46 27" src="https://github.com/user-attachments/assets/46410109-c3f9-4126-9233-1323b6212311">